src/main/java/org/hubiquitus/hapi/transport/XhrTransport.java

```diff
@@ -1,297 +1,272 @@
+                            mIsConnected = true;
+                            //Start XHR polling thread when connected and authenticated
+                            mPollThread = new PollThread();
+                            mPollThread.start();
+
+                        } else {
+                            if (mTransportListener != null) {
+                                mTransportListener.onError(InternalErrorCodes.AUTHENTICATION_FAILED,
+                                        "Http status=" + (responseAuth == null ? "no response" : responseAuth.getStatus()));
+                            }
+                        }
+                    } else {
+                        if (mTransportListener != null) {
+                            mTransportListener.onError(InternalErrorCodes.CONNECTION_FAILED,
+                                    "Can't connect to host " + (responseConnect == null ? "no response" : responseConnect.getStatus()));
+                        }
+                    }
+                } catch (IOException | JSONException e) {
+                    if (mTransportListener != null) {
+                        mTransportListener.onError(InternalErrorCodes.CONNECTION_FAILED, e);
+                    }
+                }
             }
-		}).start();
-		
-	}
-
-	@Override
-	public void disconnect() {
-		isConnected = false;
-		this.transportListener.onDisconnect();
-	}
-	
-	@Override
-	public void silentDisconnect() {
-		isConnected = false;
-	}
-
-	@Override
-	protected void send(JSONObject jsonObject) throws TransportException {
-		
-		Message message = new Message();
-		message.setJsonContent(jsonObject);
-		queue.add(message);
-		
-		handleQueueMessages();
-	}
-	
-	@Override
-	public JSONObject send(String to, Object content, int timeout,
-			ResponseListener responseListener) throws TransportException {
-		
-		JSONObject jsonMessage = super.send(to, content, timeout, responseListener);
-			
-		Message message = new Message();
-		message.setJsonContent(jsonMessage);
-		message.setResponseListener(responseListener);
-		
-		queue.add(message);
-		
-		handleQueueMessages();
-		
-		return jsonMessage;
-	}
+        });
+    }
+
+    @Override
+    public void send(JSONObject jsonObject) {
+        internalSend(jsonObject);
+    }
 
     @Override
-    protected void sendHeartBeat() throws TransportException {
+    public JSONObject send(String to, Object content, int timeout, ResponseListener responseListener) {
+        JSONObject jsonMessage = super.send(to, content, timeout, responseListener);
+
+        internalSend(jsonMessage);
+        return jsonMessage;
+    }
+
+    private void internalSend(final JSONObject jsonObject) {
+        if (!isReady()) {
+            //It's normally not possible to be here, hubiquitus should check this case before calling this method
+            Log.wtf(getClass().getCanonicalName(),"try to send with not ready transport");
```

Comment:
`Log.wtf` c'est pas "What The F***" :) C'est "What a Terrible Failure", encore pire que `Log.e`.